wallet-new/README.md

```diff
@@ -1,48 +1,149 @@
 
-Currently the only way to generate an updated `swagger.json` is to run the `cardano-node` node, so that
-the updated Swagger file will be written on disk. For example:
+1. `cardano-node`
+2. `cardano-node-simple`
+
+When we start `cardano-node`, wallet functionality is activated automatically. But when we
+start `cardano-node-simple`, wallet functionality is disabled, in this case node's log will
+contain the following line:
+
+```
+Wallet is disabled, because software is built w/o it
+```
+
+## API
+
+We describe how to interact with our API via the popular [Swagger](https://swagger.io/)
+framework and format. Swagger relies on a single specifications file in a `json` format. From
+this file is derived a handful of tools.
+
+A Cardano node exposes both a Swagger file corresponding to the wallet API and a visualization
+tool that can be used for browsing the specification as well as playing with the API. As a
+first step, start a `cardano-node` using the following command:
+
+```
+$ stack exec cardano-node -- --topology=wallet-new/topology-examples/testnet.yaml \
+                             --configuration-key=mainnet_staging_short_epoch_full \
+                             --wallet-debug                                       \
+                             --rebuild-db
+```
+
+From there, you can browse the API documentation for V0 and V1 through the following URLs:
+
+- http://localhost:8090/docs/v0/index/
+- http://localhost:8090/docs/v1/index/
+
+The visualization at those URLs lets you play with the API by the mean of a _Try it out_ button
+made available for each endpoint. This will seemingly contact the node already running on your
+local machine with actual HTTP requests augmented with the parameters you provide!
+
+### HTTPS
+
+By default, wallet backend only accepts HTTPS connections:
 
 ```
-stack exec cardano-node -- --topology=wallet-new/topology-examples/testnet.yaml --configuration-key mainnet_staging_short_epoch_full --wallet-debug --rebuild-db
+$ curl localhost:8090/docs/v1/index/index.html
+This server only accepts secure HTTPS connections.
 ```
 
-Running the command above *from the root of the Cardano project* will store an updated `swagger.json` into
-`wallet-new/spec`.
+We should provide our `ca.crt`:
+
+```
+$ curl --cacert scripts/tls-files/ca.crt https://localhost:8090/docs/v1/index/index.html
+```
+
+But if we launch a node with `--wallet-debug` option, we can send simple `http`-requests.
+
+### Swagger Specification
+
+If needed, you can access the corresponding raw Swagger specification files via these URLs:
+
+- http://localhost:8090/docs/v0/swagger.json
+- http://localhost:8090/docs/v1/swagger.json
 
-### Playing with the API
+Please note that the latest specification should be stored in [`wallet-new/spec`](https://github.com/input-output-hk/cardano-sl/tree/develop/wallet-new/spec)
+directory. So if you made any changes in API - please update `/wallet-new/spec/swagger.json`
+as well.
```

Comment:
Oui! 
`.gitignore` won't help us however on that one since it's already versionned :(  
We have to be quite careful not touching again the `wallet-new/spec` ... Would be a good idea to delete it with this PR as well to my opinion, and enforce it with subsequent PRs.

Reply:
Not a bad idea -- @denisshevchenko , would you mind purging `wallet-new/spec` once for all, so we don't fall into the same trap again?